src/components/periodic-report/handlers/sync-progress-report-to-engagement.handler.ts

```diff
@@ -0,0 +1,124 @@
+import { isNull, node, relation } from 'cypher-query-builder';
+import { DateTime } from 'luxon';
+import { Session } from '../../../common';
+import {
+  DatabaseService,
+  EventsHandler,
+  IEventHandler,
+  ILogger,
+  Logger,
+} from '../../../core';
+import { Engagement } from '../../engagement';
+import { EngagementUpdatedEvent } from '../../engagement/events';
+import { CreatePeriodicReport } from '../dto';
+import { ReportType } from '../dto/type.enum';
+import { PeriodicReportService } from '../periodic-report.service';
+
+type SubscribedEvent = EngagementUpdatedEvent;
+
+@EventsHandler(EngagementUpdatedEvent)
+export class SyncProgressReportToProject
+  implements IEventHandler<SubscribedEvent> {
+  constructor(
+    private readonly db: DatabaseService,
+    private readonly periodicReports: PeriodicReportService,
+    @Logger('periodicReport:sync-project') private readonly logger: ILogger
```

Comment:
Convention for logger names is `kebab-case`.
```suggestion
    @Logger('periodic-report:engagement-sync') private readonly logger: ILogger
```